src/profiles/models_chain_links.rs

```diff
@@ -1,46 +1,83 @@
-use crate::profiles::models_common::PubKey;
+//! Contains structs and enums related to the chain links.
+
+use crate::types::PubKey;
 use cosmwasm_std::Addr;
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
+/// Types of address encoding format.
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
-#[serde(rename_all = "snake_case")]
-pub struct ChainLink {
-    pub user: Addr,
-    pub address: ChainLinkAddr,
-    pub proof: Proof,
-    pub chain_config: ChainConfig,
-    pub creation_time: String,
+pub enum AddressType {
+    /// Address bech32 encoded.
+    #[serde(rename = "desmos.profiles.v2.Bech32Address")]
```

Comment:
If the version of profiles change to v3 from v2, `AddressType` would be not available. I prefer to remain string first, instead.

What do you think @manu0466 @bragaz ?

Reply:
I agree with you @dadamu. It will make it easier to maintain it since it already a big burden